Fixes #23063 - Add notifications on subscription API errors

tstrachota opened the pull request:
I presume that we want to show notifications on errors at other pages too so I created a common helper that dispatches the error action and adds a notification. It also standardizes what data we send with API error actions.

sharvit commented on webpack/move_to_foreman/common/helpers.js:
```diff
@@ -15,3 +17,17 @@ export const KEY_CODES = {
+  dispatch({
+    type: actionType,
+    result,
+    error,
```
Can you move the `result` and `error` to be under `payload` object?
I think we should always dispatch only `type` and `payload` and keep this as a standard.

waldenraines commented:
I wonder if we want to remove the inline alert section now that we are showing toast notifications.  It seems a little redundant now:

![screenshot from 2018-07-11 12-40-14](https://user-images.githubusercontent.com/4116405/42587093-b58e098c-8507-11e8-85e8-03a5d1b3e3e6.png)

waldenraines commented:
Is it possible that this also fixes [#24127](https://projects.theforeman.org/issues/24127)?

tstrachota commented:
Updated:
- error actions now use format: `{ type, payload: {message}}`
- error alerts from the subscription page removed

waldenraines commented:
> @waldenraines yes, it should fix that bug too

Do you want to add that to the commit message or maybe we can close [#23063](http://projects.theforeman.org/issues/23063) as a duplicate since [#24127](https://projects.theforeman.org/issues/24127) is tied to a redmine?

waldenraines commented on webpack/scenes/Subscriptions/Manifest/__tests__/manifest.fixtures.js:
```diff
@@ -210,7 +249,20 @@ export const deleteManifestFailureActions = [
+        type: 'error',
+      },
+    },
+    type: 'TOASTS_ADD',
```
I know this is just a fixture but do we want to combine all of these into a `const` so we don't have to repeat ourselves here?

tstrachota commented:
@waldenraines updated. I added test helpers for toast and failure actions. I also removed `api/fixtures.js` that's no longer used. It contained some repo request data but we're not importing it anywhere.

tstrachota commented on webpack/move_to_foreman/common/helpers.js:
```diff
@@ -1,11 +1,30 @@
-    return `/${base}?search=${searchQuery}`;
-  },
+export const getResponseErrorMsg = ({ data }) =>
+  data && (data.displayMessage || data.message || data.error);
```
@waldenraines I saw your question about whether this handles list of errors in email (not sure why it disappeared from here):
Are there any responses that return **only** list of errors? From what I saw there's always a `displayMessage` with them and it seemed sufficient to display just that. We can check for list of errors and  send a notification for each of them. What do you think?

tstrachota replied:
Per a discussion on irc:
We do need to process `errors` and they should take precedence over `displayMessage`.

tstrachota commented:
Updated, `errors` in API responses now take precedence over `displayMessage`.
